src/mapper/mapper_test.go

```diff
@@ -38,3 +43,113 @@ func createNamespace(name string, labels map[string]string) *corev1.Namespace {
 		},
 	}
 }
+
+func TestUpdateNamespace(t *testing.T) {
+	t.Run("Add to namespace", func(t *testing.T) {
+		labels := map[string]string{"test": "selector"}
+		dk := createTestDynakubeWithMultipleFeatures("dk-test", labels, nil)
+		namespace := createNamespace("test-namespace", labels)
+
+		updated, err := updateNamespace(namespace, &dynatracev1beta1.DynaKubeList{Items: []dynatracev1beta1.DynaKube{*dk}})
+
+		require.NoError(t, err)
+		require.True(t, updated)
+		assert.Equal(t, 2, len(namespace.Labels))
+	})
+	t.Run("Overwrite stale entry in labels", func(t *testing.T) {
```

Comment:
that doesn't look like it checks for the label to have actually changed

Reply:
done